### PR TITLE
Item Magnet -> Dowsing Machine

### PR DESCRIPTION
**Pull request by Ultima1990**
To free up the Magnet for other uses.

With bonus NPC for the funny.

Hoping for some hero to bring in a higher quality Dowsing Machine sprite.

### CODE REVIEW

File: src/scripts/Update.ts [hunk 1]
```diff
@@ -746,6 +746,11 @@ class Update implements Saveable {
 
             // Add AZ TemporaryBattle
             saveData.statistics.temporaryBattleDefeated = Update.moveIndex(saveData.statistics.temporaryBattleDefeated, 0);
+
+            // Give the players Dowsing Machines in place of Item Magnets
+            playerData._itemList.Dowsing_machine = playerData._itemList.Item_magnet;
+            delete playerData._itemList.Item_magnet;
+            delete localStorage.mine;
```
Comment (Jaaslet) [inline src/scripts/Update.ts:753]: I think localStorage.mine was something else, that should not have been copied. At least it works fine without.

**Review by Jaaslet — CHANGES_REQUESTED** [review]
You are missing a battle frontier reward. Else it works. I have one more small comment:

File: src/scripts/Update.ts (hunk 1 above)
Comment (Ultima1990) [reply src/scripts/Update.ts]: I fix!

**Review by Jaaslet — APPROVED** [review]
Looks good. Thanks for the PR

**Pull request merged**